core/coreunix/add.go

```diff
@@ -132,6 +133,13 @@ func (adder *Adder) add(reader io.Reader) (ipld.Node, error) {
 	if adder.Trickle {
 		nd, err = trickle.Layout(db)
 	} else {
+		if db.IsThereMetaData() && !db.IsMetaDagBuilt() {
+			err := balanced.BuildMetadataDag(db)
```

Comment:
This is balance specific logic, is it possible to do it inside `Layout()` function? I see some tests use the same logic. If not possible maybe a new flag on `Layout()'? Maybe there is a reason, let me know, thx